### PR TITLE
[TS] LPS-123216 Fix NPE in FacetBucketUtil

### PR DESCRIPTION
**Pull request by lipusz**
https://issues.liferay.com/browse/LPS-123216

Background: https://issues.liferay.com/browse/LPS-123216?focusedCommentId=2310468&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2310468

Caused by: https://issues.liferay.com/browse/LPS-123216?focusedCommentId=2311620&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2311620

From André:
> Well, regardless of the User document, it does look like FacetBucketUtil has a bug. It should be more robust against undefined and other possible terms missing a bucket, instead of dying with a NullPointerException. Search should replicate this scenario with an integration test, fix it by returning false (field is not in bucket) and let U&SM figure out what they want to do with the User document.

So I'm not dealing with the `roleIds` part, though we should probably report this to the U&SM team as the current logic can result in a lots of documents fetched back from the search engine unnecessarily.

### CODE REVIEW

**Comment by lipusz** [comment]
ci:test:sf

**Comment by lipusz** [comment]
ci:test:relevant

**Comment by lipusz** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/FacetBucketUtil.java [hunk 1]
```diff
@@ -37,7 +38,9 @@ public static boolean isFieldInBucket(
 
 			String value = field.getValue();
```
Comment (BryanEngler) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/FacetBucketUtil.java:39]: do you think `value` would also need a null check?

Comment (lipusz) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/FacetBucketUtil.java:39]: Does our Field impl permits null values? 🤔 @BryanEngler

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/FacetBucketUtil.java [hunk 2]
```diff
@@ -37,7 +38,9 @@ public static boolean isFieldInBucket(
...
 
-			if ((lower.compareTo(value) <= 0) && (value.compareTo(upper) < 0)) {
+			if (Validator.isNotNull(lower) && (lower.compareTo(value) <= 0) &&
+				Validator.isNotNull(upper) && (value.compareTo(upper) < 0)) {
```
Comment (BryanEngler) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/FacetBucketUtil.java:42]: possible NPE at `value.compareTo`?

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/FacetBucketUtil.java (hunk 1 above)
Comment (BryanEngler) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/FacetBucketUtil.java:39]: i think so, but if the field didnt have a value, it probably wouldnt be in a facet right? so were probably fine

Comment (lipusz) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/FacetBucketUtil.java:39]: I think so. But I can check tomorrow if you prefer to.

**Comment by BryanEngler** [comment]
lgtm, can forward once tests pass

**Comment by lipusz** [comment]
Thx Bryan!

**Comment by lipusz** [comment]
ci:forward

**Comment by lipusz** [comment]
ci:forward

**Comment by lipusz** [comment]
:white_check_mark: Failure in relevant is known, https://issues.liferay.com/browse/LPS-123052

**Comment by lipusz** [comment]
ci:forward

**Pull request closed without merging**